products/error_tracking/frontend/utils.ts

```diff
@@ -106,25 +108,32 @@ export class ResolvedDateRange {
         }
     }
 
-    static fromDateRange(dateRange: DateRange): ResolvedDateRange {
-        return new ResolvedDateRange(resolveDate(dateRange.date_from), resolveDate(dateRange.date_to))
+    static fromDateRange(dateRange: DateRange, precision?: DateRangePrecision): ResolvedDateRange {
+        const fromOffset = precision
+            ? dayjs().subtract(precision.value, precision.unit).startOf(precision.unit)
+            : dayjs()
+        const toOffset = precision ? dayjs().add(precision.value, precision.unit).endOf(precision.unit) : dayjs()
+        return new ResolvedDateRange(
+            resolveDate(fromOffset, dateRange.date_from),
+            resolveDate(toOffset, dateRange.date_to)
+        )
```

Comment:
This always makes the range wider than specified, right? I think that's the right approach when playing with the specified range.

There's a chance that a user is refreshing the page looking for new issues / exceptions. I wonder if we should clamp it at a shorter range (30s or 1 minute) for `dateTo` values. Maybe the 'Reload' button is enough because I believe that hard refreshes because of the `force_blocking` param

Reply:
Actually... what is the default `refresh` param on all of these queries?

Reply:
Nevermind it's `blocking` so that will hit the cache if this date range is consistent 👌